src/librustc/ty/context.rs

```diff
@@ -1472,21 +1471,13 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
     /// because that method has a narrower effect that can be toggled
     /// off via a separate `-Z` flag, at least for the short term.
```

Comment:
add this to your to-do list in the description:  get rid of one of the two methods `allow_bind_by_move_patterns_with_guards` vs `check_for_mutation_in_guard_via_ast_walk`. (I used to have a `-Z` flag that toggle check_for_mutation_in_guard_via_ast_walk` but clearly that is not here any more...)